---
title: "Translation on Demand"
format: dashboard
logo: "CECAD-Logo-24-white.png"
theme: minty
embed-resources: true
scrollable: true
nav-buttons:
      - icon: github
        href: https://github.com/eliasschwall/translation_on_demand
---

```{r set up}
#| label: set up chunk

library(tidyverse)
library(here)
library(DT)
library(gt)
```

# Data Overview 

## Row 
```{r update valuebox}
#| content: valuebox
#| title: "Latest update"
list(
  icon = "clock",
  color = "secondary",
  value = Sys.Date()
)
```

::: {.card title="Data Source"}
The data used for this project is from Yang et al., 2019: [Multi-omic Profiling Reveals Dynamics of the Phased Progression of Pluripotency](https://www.sciencedirect.com/science/article/pii/S2405471219301152?via%3Dihub).

- [Table S2](https://ars.els-cdn.com/content/image/1-s2.0-S2405471219301152-mmc3.xlsx): Temporal Proteomic Data (0min, 30min, 1h, 6h, 12h, 24h, 36h, 48h, and 72h)
- [Table S3](https://ars.els-cdn.com/content/image/1-s2.0-S2405471219301152-mmc4.xlsx): Temporal Transcriptomic (RNA-Seq) Data (0min, 1h, 6h, 12h, 24h, 36h, 48h, and 72h)

For the purpuse of our project we realigned the count data for the transcriptomic data with the M11 mouse genome using STAR. 
:::

## Row

```{r displaying source data proteome}
#| title: Proteome Data Head

proteomic_yang_data <- here::here("data/processed/yang_paper", "yang_proteomics.rds") %>% readRDS()

proteomic_yang_data %>% 
  head() %>% 
  gt()
```

## Row

```{r displaying source data transcriptome}
#| title: Transcriptome Data Head

transcriptomic_yang_data <- here::here("data/raw/yang_paper", "transcriptomic_counts.csv") |> read.csv()

transcriptomic_yang_data %>% 
  head() %>% 
  gt()
```

# ToD Filtering 

## Row {.tabset title="ToD candidate genes"}
```{r}
#| title: 0h to 1h
ToD_candidates_list <- readRDS(here::here("data/processed", "ToD_candidates_list.rds"))

datatable(ToD_candidates_list$`1h_vs_0h`, 
          extensions = 'Buttons',
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'ToD_candidates_0h_to_1h'),
              list(extend = 'excel', title = 'ToD_candidates_0h_to_1h')
            )
          )
)
```

```{r}
#| title: 1h to 6h

datatable(ToD_candidates_list$`6h_vs_1h`, 
          extensions = 'Buttons',
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'ToD_candidates_1h_to_6h'),
              list(extend = 'excel', title = 'ToD_candidates_1h_to_6h')
            )
          )
)
```

```{r}
#| title: 6h to 12h

datatable(ToD_candidates_list$`12h_vs_6h`, 
          extensions = 'Buttons',
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'ToD_candidates_6h_to_12h'),
              list(extend = 'excel', title = 'ToD_candidates_6h_to_12h')
            )
          )
)
```

```{r}
#| title: 12h to 24h

datatable(ToD_candidates_list$`24h_vs_12h`, 
          extensions = 'Buttons',
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'ToD_candidates_12h_to_24h'),
              list(extend = 'excel', title = 'ToD_candidates_12h_to_24h')
            )
          )
)
```

```{r}
#| title: 24h to 36h

datatable(ToD_candidates_list$`36h_vs_24h`, 
          extensions = 'Buttons',
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'ToD_candidates_24h_to_36h'),
              list(extend = 'excel', title = 'ToD_candidates_24h_to_36h')
            )
          )
)
```

```{r}
#| title: 36h to 48h

datatable(ToD_candidates_list$`48h_vs_36h`, 
          extensions = 'Buttons',
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'ToD_candidates_36h_to_48h'),
              list(extend = 'excel', title = 'ToD_candidates_36h_to_48h')
            )
          )
)
```

```{r}
#| title: 48h to 72h

datatable(ToD_candidates_list$`72h_vs_48h`, 
          extensions = 'Buttons',
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'ToD_candidates_48h_to_72h'),
              list(extend = 'excel', title = 'ToD_candidates_48h_to_72h')
            )
          )
)
```

# GO Enrichment 

# TF Analysis

```{r}
#| title: TF Prot-RNA-LFC-Ratio Over Time
#| label: looking at the temporal behavior of ToD TFs

mouse_TFs <- here("data/reference", "Mus_musculus_TF.txt") %>% read_tsv()
matched_list <- readRDS(here::here("data/processed", "matched_list.rds"))

ToDCs_TFs <- lapply(ToD_candidates_list, function(time_interval){
  intersect(time_interval$mgi_symbol,mouse_TFs$Symbol)
}) 

# merge all time points in one df
merged_matched <- Reduce(function(x, y) merge(x, y, by = "mgi_symbol", all = TRUE), matched_list) %>%
  janitor::clean_names() %>%
  select(mgi_symbol, colnames(.)[str_detect(colnames(.), "diff")]) %>%
  setNames(c("gene", "h0_h1", "h1_h6", "h6_h12", "h12_h24", "h24_h36", "h36_h48", "h48_h72"))

# filtering this df for all ToD TFs
ToDCs_TFs_all <- c(
  ToDCs_TFs$`1h_vs_0h`,
  ToDCs_TFs$`6h_vs_1h`,
  ToDCs_TFs$`12h_vs_6h`,
  ToDCs_TFs$`24h_vs_12h`,
  ToDCs_TFs$`36h_vs_24h`,
  ToDCs_TFs$`48h_vs_36h`,
  ToDCs_TFs$`72h_vs_48h`
)

merged_matched_TFs <- merged_matched %>% filter(gene %in% ToDCs_TFs_all)

# Transform the data for ggplot
data_long <- merged_matched_TFs %>%
  pivot_longer(cols = -gene, names_to = "timepoint", values_to = "value") %>%
  mutate(timepoint = recode(timepoint,
    h0_h1 = "0-1h",
    h1_h6 = "1-6h",
    h6_h12 = "6-12h",
    h12_h24 = "12-24h",
    h24_h36 = "24-36h"
  ))

# Create the plot
p <- ggplot(data_long, aes(x = timepoint, y = value, group = gene, color = gene)) +
  geom_line() +
  geom_point() +
  labs(y = "Prot-RNA-LFC-Ratio") +
  geom_hline(yintercept = 1, linetype = "dashed", color = "red") +
  annotate("text", x = 7.5, y = 1.5, label = "ToDC", color = "red")+
theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) 

# Convert to plotly
p_plotly <- plotly::ggplotly(p, tooltip = c("gene", "value"))

# Set all traces to be turned off by default
for (i in 1:length(p_plotly$x$data)) {
  if (i != 105 && i != 106) {  # Replace y_intercept_index with the actual index of your y-intercept line
    p_plotly$x$data[[i]]$visible <- "legendonly"
  }
}

# Show the plot
p_plotly
```

# Motif Analysis 

## Row

```{r}
#| title: RNA Binding Protein Target Motifs Enriched in ToD Candidates

# Group the data by time_interval
grouped_data <- readRDS(here::here("results/meme_analysis", "RBP_motif_enrichmets_df.rds")) %>%
  group_by(time_interval)

# Create the DT table
datatable(grouped_data, 
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'RBP_Motif_Enrichments_in_ToD_candidates'),
              list(extend = 'excel', title = 'RBP_Motif_Enrichments_in_ToD_candidates')
            )
          )
)
```

## Row {.tabset title="ToD candidates carrying RBP motifs"} 
```{r}
#| title: 0h to 1h 5' UTRs

ToD_genes_carrying_RBP_motifs <- here::here("results/meme_analysis", "ToD_genes_carrying_RBP_motifs.rds") %>% 
  readRDS()

ToD_genes_carrying_RBP_motifs$`1h_vs_0h`$utr_5 %>%
  group_by(motif) %>%
  datatable(., 
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'ToD_genes_RBP_motifs_0h_to_1h_5_utr'),
              list(extend = 'excel', title = 'ToD_genes_RBP_motifs_0h_to_1h_5_utr')
            )
          )
)

```

```{r}
#| title: 0h to 1h 3' UTRs

ToD_genes_carrying_RBP_motifs$`1h_vs_0h`$utr_3 %>%
  group_by(motif) %>%
  datatable(., 
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'ToD_genes_RBP_motifs_0h_to_1h_3_utr'),
              list(extend = 'excel', title = 'ToD_genes_RBP_motifs_0h_to_1h_3_utr')
            )
          )
)
```

```{r}
#| title: 1h to 6h 5' UTRs

ToD_genes_carrying_RBP_motifs$`6h_vs_1h`$utr_5 %>%
  group_by(motif) %>%
  datatable(., 
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'ToD_genes_RBP_motifs_1h_to_6h_5_utr'),
              list(extend = 'excel', title = 'ToD_genes_RBP_motifs_1h_to_6h_5_utr')
            )
          )
)
```

```{r}
#| title: 6h to 12h 5' UTRs

ToD_genes_carrying_RBP_motifs$`12h_vs_6h`$utr_5 %>%
  group_by(motif) %>%
  datatable(., 
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'ToD_genes_RBP_motifs_6h_to_12h_5_utr'),
              list(extend = 'excel', title = 'ToD_genes_RBP_motifs_6h_to_12h_5_utr')
            )
          )
)
```

```{r}
#| title: 12h to 24h 5' UTRs

ToD_genes_carrying_RBP_motifs$`24h_vs_12h`$utr_5 %>%
  group_by(motif) %>%
  datatable(., 
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'ToD_genes_RBP_motifs_12h_to_24h_5_utr'),
              list(extend = 'excel', title = 'ToD_genes_RBP_motifs_12h_to_24h_5_utr')
            )
          )
)
```

```{r}
#| title: 24h to 36h 5' UTRs

ToD_genes_carrying_RBP_motifs$`36h_vs_24h`$utr_5 %>%
  group_by(motif) %>%
  datatable(., 
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'ToD_genes_RBP_motifs_24h_to_36h_5_utr'),
              list(extend = 'excel', title = 'ToD_genes_RBP_motifs_24h_to_36h_5_utr')
            )
          )
)
```

```{r}
#| title: 36h to 48h 5' UTRs

ToD_genes_carrying_RBP_motifs$`48h_vs_36h`$utr_5 %>%
  group_by(motif) %>%
  datatable(., 
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'ToD_genes_RBP_motifs_36h_to_48h_5_utr'),
              list(extend = 'excel', title = 'ToD_genes_RBP_motifs_36h_to_48h_5_utr')
            )
          )
)
```

```{r}
#| title: 36h to 48h 3' UTRs

ToD_genes_carrying_RBP_motifs$`48h_vs_36h`$utr_3 %>%
  group_by(motif) %>%
  datatable(., 
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'ToD_genes_RBP_motifs_36h_to_48h_3_utr'),
              list(extend = 'excel', title = 'ToD_genes_RBP_motifs_36h_to_48h_3_utr')
            )
          )
)
```

```{r}
#| title: 48h to 72h 5' UTRs

ToD_genes_carrying_RBP_motifs$`72h_vs_48h`$utr_5 %>%
  group_by(motif) %>%
  datatable(., 
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'ToD_genes_RBP_motifs_48h_to_72h_5_utr'),
              list(extend = 'excel', title = 'ToD_genes_RBP_motifs_48h_to_72h_5_utr')
            )
          )
)
```

```{r}
#| title: 48h to 72h 3' UTRs

ToD_genes_carrying_RBP_motifs$`72h_vs_48h`$utr_3 %>%
  group_by(motif) %>%
  datatable(., 
          options = list(
            dom = 'Bfrtip',
            buttons = list(
              list(extend = 'csv', title = 'ToD_genes_RBP_motifs_48h_to_72h_3_utr'),
              list(extend = 'excel', title = 'ToD_genes_RBP_motifs_48h_to_72h_3_utr')
            )
          )
)
```